test/e2e_node/builder/build.go

```diff
@@ -47,6 +49,11 @@ func BuildGo() error {
 	}
 	targets := strings.Join(buildTargets, " ")
 	cmd := exec.Command("make", "-C", k8sRoot, fmt.Sprintf("WHAT=%s", targets))
+	if IsBuildForArm64() {
```

Comment:
I think this should check `useDockerizedBuild` flag, not build for ARM. There are other usecases where dockerized build will be helpful and it shouldn't be restricted to just ARM.

I would suggest we use dockerized build if the dockerized flag is set. If ARM is set too as `target-build-arch` then, set `KUBE_BUILD_PLATFORMS`.

Reply:
I guess I can break this PR into two. In this PR, I want to highlight ARM only supports dockerized build. This one will focus on enable tests running on ARM machine, without the risk of introducing potential regression to x86 run. A follow-up PR will  enable x86 with useDockerizedBuild, and ensure there's no test regression with or without the `--use-dockerized-flag`

Reply:
I see... I would think if we set --use-dockerized-flag to false by default and use the old behavior if it's false, I think it should be safe: running the presubmit node e2e job would give good signal that nothing would be broken.

If the goal is to highlight ARM only supports dockerized build, maybe instead of defaulting, it would make sense to just return an error that specifies that for ARM, we need dockerized build. So something like:


```
if targetBuildArch == "arm64" && !isDockerizedBuild { 
    ... return error ...
}

if !isDockerizedBuild {
   ... do what we do today ...
} else {
   cmd = exec.Command(filepath.Join(k8sRoot, "build/run.sh"), "make", fmt.Sprintf("WHAT=%s", targets), "KUBE_BUILD_PLATFORMS=linux/" + targetBuildArch)
}
```

Reply:
Sounds good to me. I updated the semantics a bit for targetBuildArch (use 'linux/arm64' instead of 'arm64') so that `GetK8sBuildOutputDir()` doesn't need too much changes for KUBE_BUILD_PLATFORMS. I think now it should be pretty generic.